### PR TITLE
MOSIP-11739 code changes for MOSIP-15168

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package io.mosip.registration.processor.core.code;
 
 
+// TODO: Auto-generated Javadoc
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java]: Why is this TODO added ?

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java [hunk 2]
```diff
@@ -1245,7 +1245,11 @@
 
 	RPR_WIA_UNKNOWN_WORKFLOW_ACTION(PlatformConstants.RPR_WORKFLOW_INTERNAL_ACTION + "002", "Unknown Exception"),
 
-	RPR_WIA_SUBWORKFLOW_NOT_FOUND(PlatformConstants.RPR_WORKFLOW_INTERNAL_ACTION + "003", "SubWorkflow not found"),
+	RPR_WIA_ADDITIONALINFOPROCESS_NOT_FOUND(PlatformConstants.RPR_WORKFLOW_INTERNAL_ACTION + "003",
+			"AdditionalInfoProcess not found"),
+
+	RPR_WIA_ADDITIONALINFOPROCESS_CANNOT_REQUEST(PlatformConstants.RPR_WORKFLOW_INTERNAL_ACTION + "004",
+			"AdditionalInfoProcess cannot be  Paused for one more additional info so it is failed"),
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java]: remove the extra space before Paused and make pause with small p

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformSuccessMessages.java [hunk 3]
```diff
@@ -143,9 +143,9 @@
 	PAUSE_AND_REQUEST_ADDITIONAL_INFO(PlatformConstants.RPR_CAMEL_BRIDGE_MODULE + "005",
 			"packet paused and request additional info"),
 
-	PACKET_RESUME_PARENT_FLOW(PlatformConstants.RPR_CAMEL_BRIDGE_MODULE + "006", "packet parent flow resume"),
+	PACKET_RESTART_PARENT_FLOW(PlatformConstants.RPR_CAMEL_BRIDGE_MODULE + "007", "packet parent flow restart"),
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformSuccessMessages.java]: why 006 is skipped ?  Also make the message to say "Packet parent flow restart initiated"

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java [hunk 4]
```diff
@@ -232,10 +259,34 @@ private void processCompleteAsFailed(WorkflowInternalActionDTO workflowInternalA
...
+		if (subWorkflowDtos != null && !subWorkflowDtos.isEmpty()) {
+			Map<String, String> tags = new HashMap<String, String>();
+			tags.put(workflowInternalActionDTO.getReg_type() + "_FLOW_STATUS",
+					RegistrationStatusCode.FAILED.toString());
+			packetManagerService.addOrUpdateTags(workflowInternalActionDTO.getRid(),
+					tags);
+			SubWorkflowDto subWorkflowDto = subWorkflowDtos.get(0);
+			InternalRegistrationStatusDto mainFlowregistrationStatusDto = registrationStatusService
+					.getRegistrationStatus(
+					subWorkflowDto.getRegId(), subWorkflowDto.getParentProcess(), subWorkflowDto.getParentIteration());
```
Comment (vishwa-vyom) [inline registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: pass null for now for the 4th argument everywhere for getRegistrationStatus method is called with subWorkflow details

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java [hunk 5]
```diff
@@ -81,6 +81,20 @@ public boolean matches(Exchange exchange) {
...
+		JsonObject json = new JsonObject(message);
+		WorkflowInternalActionDTO workflowEventDTO = new WorkflowInternalActionDTO();
+		workflowEventDTO.setRid(json.getString(JsonConstant.RID));
+		workflowEventDTO.setActionCode(WorkflowInternalActionCode.COMPLETE_AS_REJECTED_WITHOUT_PARENT_FLOW.toString());
+		workflowEventDTO
+				.setActionMessage(PlatformSuccessMessages.PACKET_COMPLETE_AS_REJECTED_WITHOUT_PARENT_FLOW.getMessage());
+		workflowEventDTO.setReg_type(json.getString(JsonConstant.REGTYPE));
+		workflowEventDTO.setIteration(json.getInteger(JsonConstant.ITERATION));
+		workflowEventDTO.setSource(json.getString(JsonConstant.SOURCE));
+		exchange.getMessage().setBody(objectMapper.writeValueAsString(workflowEventDTO));
```
[the remark this answers is not in the document]
Comment (sowmya695) [reply registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/WorkflowCommandPredicate.java]: Fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java (hunk 1 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/code/WorkflowInternalActionCode.java]: while adding java doc for new code its added ,fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java (hunk 2 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java]: fixed

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformSuccessMessages.java (hunk 3 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformSuccessMessages.java]: as resume code removed,i didnt notice the errorCode.fixed

File: registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java (hunk 4 above)
Comment (sowmya695) [reply registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/WorkflowInternalActionVerticle.java]: fixed

**Pull request merged**